test/integration/full/preload/preload.js

```diff
@@ -0,0 +1,218 @@
+/* global axe, Promise */
+describe('preload integration test', function() {
+	'use strict';
+
+	var origAxios;
+	var origAuditRun;
+
+	function overridedCheckEvaluateFn(node, options, virtualNode, context) {
+		// populate the data here which is asserted in tests
+		this.data(context);
+		return true;
+	}
+
+	before(function(done) {
+		function start() {
+			// cache originals
+			origAuditRun = axe._audit.run;
+			if (axe.imports.axios) {
+				origAxios = axe.imports.axios;
+			}
+			// load custom rule
+			// load custom check
+			// the evaluate in check sets the data object
+			// which is used for assertion in tests below
+			axe._load({
+				rules: [
+					{
+						// this rule is not preload dependent
+						// and can run immediately
+						id: 'run-now-rule',
+						selector: 'div#run-now-target',
+						any: ['check-context-exists']
+					},
+					{
+						// this rule requires preload
+						// and will run after preload assets are ready
+						id: 'run-later-rule',
+						selector: 'div#run-later-target',
+						any: ['check-context-has-assets'],
+						preload: true
```

Comment:
For the sake of future proofing this, I think we should specify what this needs to preload: `preload: 'cssom'`.